kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/ALBookingRMAdapter.java

```diff
@@ -0,0 +1,534 @@
+            emailIdEt = itemView.findViewById(R.id.emailIdEt);
+            contactNumberEt = itemView.findViewById(R.id.contactNumberEt);
+            submitAction = itemView.findViewById(R.id.submitDetails);
+
+            titleList = new ArrayList<>();
+            titleList.add("Title *");
+            titleList.add("Mr.");
+            titleList.add("Ms.");
+            titleList.add("Mrs");
+
+            ArrayAdapter<String> adapter = new ArrayAdapter<String>(context, android.R.layout.simple_spinner_item, titleList);
+            titleSpinner.setAdapter(adapter);
+        }
+    }
+
+    private class GuestCountHolder extends RecyclerView.ViewHolder {
+        TextView adultCountTv, childCountTv;
+        Button adultCountDecBt, childCountDecBt, adultCountIncrementBt, childCountIncrementBt;
+        TextView roomDetailTv;
+        TextView addRoomTv, removeRoomTv, doneTv;
+        LinearLayout selectionActionLayout;
+        LinearLayout selectionRootLayout;
+
+        public GuestCountHolder(View itemView) {
+            super(itemView);
+
+            adultCountTv = itemView.findViewById(R.id.adultCountTv);
+            childCountTv = itemView.findViewById(R.id.childCountTv);
+            adultCountIncrementBt = itemView.findViewById(R.id.adultCountIncrementBt);
+            childCountIncrementBt = itemView.findViewById(R.id.childCountIncrementBt);
+            adultCountDecBt = itemView.findViewById(R.id.adultCountDecBt);
+            childCountDecBt = itemView.findViewById(R.id.childCountDecBt);
+            roomDetailTv = itemView.findViewById(R.id.alRoomDetailsTv);
+            addRoomTv = itemView.findViewById(R.id.addRoomTv);
+            removeRoomTv = itemView.findViewById(R.id.removeRoomTv);
+            doneTv = itemView.findViewById(R.id.doneButtonTv);
+            selectionActionLayout = itemView.findViewById(R.id.actionLayout);
+            selectionRootLayout = itemView.findViewById(R.id.rootSelectionLayout);
+        }
+    }
+    //holder classes <<<
+
+    private void setGuestInfoClickListeners(final GuestCountHolder holder, final int position) {
+        holder.adultCountDecBt.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int count = Integer.parseInt(holder.adultCountTv.getText().toString());
+                ALGuestCountModel guestModel = guestList.get(position);
+                if (count > 1) {
+                    holder.adultCountTv.setText(String.valueOf(count - 1));
+                    guestModel.setNoOfAdults(String.valueOf(count - 1));
+                }
+            }
+        });
+
+        holder.adultCountIncrementBt.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int count = Integer.parseInt(holder.adultCountTv.getText().toString());
+                ALGuestCountModel guestModel = guestList.get(position);
+                if (count < 5) {
```

Comment:
whats the purpose of 5? can this be moved to a constant?

Reply:
denotes max guest limit, moved